strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout-provider/src/main/java/org/carlspring/strongbox/providers/layout/Maven2LayoutProvider.java

```diff
@@ -185,6 +186,8 @@ public void delete(String storageId,
             // Otherwise, this is either not an artifact directory, or not a valid Maven artifact
         }
 
+        closeIndex(repositoryPath);
```

Comment:
Note for myself: it has to be moved out of here (org.carlspring.strongbox.providers.layout.Maven2LayoutProvider#delete)